iree/turbine/kernel/wave/decompose_reduce_ops.py

```diff
@@ -67,11 +65,11 @@ def determine_shuffle_config(
     offset = access_pattern.start.subs({k: 0 for k in ignore})
     offset = subs_idxc(offset)
     offset_table = [offset.subs({THREAD_0: i}) for i in range(subgroup_size)]
-    # Determine the thread ids participating in the shuffle.
+    unique_offsets = list(dict.fromkeys(offset_table))
+    cluster_size = len(set(offset_table))
```

Comment:
this is neat! so we count how many unique threads there are in a layout dimension? can we actually add a comment here to make it easier to understand for future selves?

Reply:
actually, is this not the same as len(unique_offsets)?

Reply:
The mistaken the original code was making was that the cluster size was being determined from the index. However the cluster size represents the number of threads that are participating in the shuffle and I don't think that can be queried directly. I will add a comment. Yes it is the same as len(unique_ offsets), will change.